pyproject.toml

```diff
@@ -0,0 +1,236 @@
+name = "pycaret"
+version = "3.4.0"
+description = "PyCaret - An open source, low-code machine learning library in Python."
+readme = "README.md"
+keywords = [
+  "data-mining",
+  "data-science",
+  "classification",
+  "clustering",
+  "detection",
+  "forecasting",
+  "machine-learning",
+  "regression",
+  "scikit-learn",
+  "time-series",
+]
+license = { file = "LICENSE" }
+maintainers = [
+  { name = "pycaret developers" },
+]
+# for full credits see contributors.md
+authors = [
+    { name = "pycaret developers" },
+]
+requires-python = ">=3.9,<3.13"
+classifiers = [
+  "Intended Audience :: Developers",
+  "Intended Audience :: Science/Research",
+  "License :: OSI Approved :: MIT License",
+  "Operating System :: MacOS",
+  "Operating System :: Microsoft :: Windows",
+  "Operating System :: POSIX",
+  "Operating System :: Unix",
+  "Programming Language :: Python",
+  "Programming Language :: Python :: 3 :: Only",
+  "Programming Language :: Python :: 3.9",
+  "Programming Language :: Python :: 3.10",
+  "Programming Language :: Python :: 3.11",
+  "Programming Language :: Python :: 3.12",
+  "Topic :: Scientific/Engineering",
+  "Topic :: Scientific/Engineering :: Artificial Intelligence",
+  "Topic :: Software Development",
+]
+# core dependencies of pycaret
+# this set should be kept minimal!
+dependencies = [
+  # Base
+  "ipython>=5.5.0",
+  "ipywidgets>=7.6.5",
+  "tqdm>=4.62.0",
+  "numpy>=1.21, <1.27",
+  "pandas>=2.2.0",  # latest sktime makes tests fail if pandas 2.1
+  "jinja2>=3",
+  "scipy>=1.6.1,<=1.11.4",  # to fix later (
+  "joblib>=1.2.0,<1.4",  # joblib<1.2.0 is vulnerable to Arbitrary Code Execution
+  "scikit-learn>1.4.0",
+  "pyod>=1.1.3",  # 1.1.3 to support scikit-learn 1.4
+  "imbalanced-learn>=0.12.0",  # >=0.12.0 to support scikit-learn 1.4
+  "category-encoders>=2.4.0",
+  "lightgbm>=3.0.0",
+  "numba>=0.55.0",
+  "requests>=2.27.1",  # Required by pycaret.datasets
+  "psutil>=5.9.0",
+  "markupsafe>=2.0.1",  # Fixes Google Colab issue
+  "importlib_metadata>=4.12.0",
+  "nbformat>=4.2.0",
+  "cloudpickle",
+  "deprecation>=2.1.0",
+  "xxhash",
+  "wurlitzer; platform_system != 'Windows'",
+  "trio>=0.22.0,<0.25.0",  # fixes problems about third-party packages, remove after httpcore 1.05
+  "setuptools; python_version>='3.12'",
+  # Plotting
+  "matplotlib<3.8.0",  # stem(..., use_line_collection=False) is no longer supported.
+  "mljar-scikit-plot",  # this is a updated version of original scikit-plot
+  "yellowbrick>=1.4",
+  "plotly>=5.14.0",
+  "kaleido>=0.2.1",
+  "schemdraw==0.15",  # 0.16 only supports Python >3.8
+  "plotly-resampler>=0.8.3.1",
+  # Time-series
+  "statsmodels>=0.12.1",
+  "sktime",  # sktime adds support for scikit-learn 1.5 in version sktime 0.31
+  "tbats>=1.1.3",
+  "pmdarima>=2.0.4",
+]
+
+[project.optional-dependencies]
+# there are the following dependency sets:
+# - full - all soft dependencies
+# - analysis
+# - models
+# - tuners
+# - mlops
+# - parallel
+# - dev
+# - test
```

Comment:
Thank you for staging this. You may also want to absorb the gist of that patch into this one?

- GH-4126

Reply:
This PR only does a pure refactor.

For changing the dep sets, I think we need to isolate the intended soft deps in the code first, or the package may break!

See more details here: https://github.com/pycaret/pycaret/pull/4126#pullrequestreview-2604481275

Reply:
GH-4129 is much better. Thanks!

Reply:
I will give you maintainer access to the fork, so you can do both.